server/validation/validateJWT.js

```diff
@@ -0,0 +1,15 @@
+module.exports = function (req, jwt) {
+	const authHeader = req.headers.authorization;
+	if (authHeader) {
+		const token = authHeader.split(' ')[1];
+		try {
+			const result = jwt.verify(token, process.env.JWT_TOKEN);
+			if (result.username === req.cookies.username) {
```

Comment:
Why would you need to compare the username here? What is the evidence that JWT is valid?

Reply:
For things like deleting a post, it's also useful to know if the real username of the user is the same as what the user claims his username is. Then, I can just use the username in his cookies if I use this kind of validation.